flasc/energy_ratio/energy_ratio.py

```diff
@@ -105,28 +105,43 @@ def _compute_energy_ratio_single(df_,
     df_ = add_ws_bin(df_, ws_cols, ws_step, ws_min, ws_max, remove_all_nulls=remove_all_nulls)
     df_ = add_wd_bin(df_, wd_cols, wd_step, wd_min, wd_max, remove_all_nulls=remove_all_nulls)
 
-    
-
     # Assign the reference and test power columns
     df_ = add_power_ref(df_, ref_cols)
     df_ = add_power_test(df_, test_cols)
 
     bin_cols_without_df_name = [c for c in bin_cols_in if c != 'df_name']
     bin_cols_with_df_name = bin_cols_without_df_name + ['df_name']
 
-    
-    
+    # Group df_
     df_ = (df_
         .filter(pl.all_horizontal(pl.col(bin_cols_with_df_name).is_not_null())) # Select for all bin cols present
         .groupby(bin_cols_with_df_name, maintain_order=True)
-        .agg([pl.mean("pow_ref"), pl.mean("pow_test"),pl.count()]) 
-        .with_columns(
-            [
-                # Get the weighting by counts
-                pl.col('count').min().over(bin_cols_without_df_name).alias('count_weight') if weight_by == 'min' else
-                pl.col('count').sum().over(bin_cols_without_df_name).alias('count_weight')
-            ]
+        .agg([pl.mean("pow_ref"), pl.mean("pow_test"),pl.count()])
+        )
+    
+    # Determine the weighting of the ws/wd bins
+
+    if df_freq_pl is None:
+        # Determine the weights per bin as either the min or sum count
+        df_ = (df_
+            .with_columns(
+                [
+                    # Get the weighting by counts
+                    pl.col('count').min().over(bin_cols_without_df_name).alias('count_weight') if weight_by == 'min' else
```

Comment:
could "count_weight" be changed to "weight" throughout the code? Especially if df_freq is used, the the weights aren't really related to counts.